modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java

```diff
@@ -375,6 +378,9 @@
     /** */
     private int qryParallelism = DFLT_QUERY_PARALLELISM;
 
+    /** Default query timeout. */
+    private long dfltQryTimeout = DFLT_QRY_TIMEOUT;
```

Comment:
It make no sense to put query settings into CacheConfiguration as cache is used for querying only as an entry point and we plan to create a new SQL API.

So, you can move this to IgniteConfiguration and mark as deprecated instantly or add a 'Ignite_Default_Query_Timeout' property to IgniteSystemProperties.